doc/upgrading.md

```diff
@@ -0,0 +1,40 @@
+# Managing Upgrades/Downgrades
+This guide describes how to upgrade your SPIRE deployment, as well as the compatibility guarantees that SPIRE users can expect.
+
+## SPIRE Versioning
+SPIRE versions are expressed as **x.y.z**, where **x** is the major version, **y** is the minor version, and **z** is the patch version, following Semantic Versioning terminology. The SPIRE project is currently pre-1.0, so minor SPIRE releases act like major releases and patch releases act like minor releases. Despite the fact that the project is pre-1.0, version compatibility guarantees are already in place.
```

Comment:
Is the version number nomenclature (i.e. major, minor, version) in the rest of the document from a pre-1.0 perspective or using the standard semver meanings? For example, when we say +/- minor version, i'm assuming we mean the semver version? once we've reached 1.0 will we need to come update this document and change minor => major, etc.?

Reply:
@azdagron would love to get your input on this one https://github.com/spiffe/spire/pull/1244/files#r342855663

Reply:
commented :) I'll resolve this convo since the other thread captures the essence